zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/TestZooKeeperConnection.java

```diff
@@ -53,10 +53,11 @@ void testPersistWatcher() throws Exception {
       _zk.writeData(path, ("datat"+i).getBytes(), -1);
       _zk.create(path+"/c2_" +i, null, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
     }
+    System.out.println("testPersistWatcher: rafter register one time listener, original listener received event count: " + get_count[0]);
+    // total number of event is 400. We will miss event now
     Assert.assertTrue(TestHelper.verify(() -> {
-          return (get_count[0].get() == 202);
-        }, TestHelper.WAIT_DURATION));
-    System.out.println("testPersistWatcher received event count: " + get_count[0]);
+      return (get_count[0].get() < 400);
```

Comment:
is there any lower bound, ie get_count[0].get > X ?

Reply:
Good point. It should have lower bound of 202.